crates/driver/src/domain/competition/order/mod.rs

```diff
@@ -190,6 +190,28 @@ impl Order {
                 .unwrap_or_default()
                 .into();
 
+        // adjust ammounts if order has volume based protocol fee
+        // this is done because solvers are unaware of protocol fees and could solve for
+        // limit prices without considering the fee, which would potentially result
+        // in a failed settlement (due to violated limit prices) once the driver
+        // tries to withold the volume based fee
+        if let Some(FeePolicy::Volume { factor }) = self.protocol_fees.first() {
+            match self.side {
+                Side::Buy => {
+                    // reduce sell amount by factor
+                    amounts.sell.amount -= amounts
+                        .sell
+                        .amount
+                        .apply_factor(*factor)
+                        .unwrap_or_default();
+                }
+                Side::Sell => {
+                    // increase buy amount by factor
+                    amounts.buy.amount +=
+                        amounts.buy.amount.apply_factor(*factor).unwrap_or_default();
```

Comment:
This seems unnecessarily inaccurate. If the orders buy amount is 100 and there is a 1% volume fee, the adjusted amount shouldn't be 100 + 1% but instead 100 / 0.99, so that the reciprocal operation (taking 1% from the result) is 100.

Reply:
Ah yes 🥇 I took completely wrong path. Fixed so no more edge case.